### ISSUE TITLE
Navigate back and forward

### ISSUE THREAD

**Issue by kappq** [epwalsh/obsidian.nvim#408]
### 🚀 The feature, motivation and pitch

One feature that I'm missing from Obsidian is the ability to navigate back and forward. Ideally it would be in the form of two new commands: `:ObsidianBack` and `:ObsidianForward`.

### Alternatives

I realize that this can somewhat be achieved using CTRL-O and CTRL-I, however it's not quite the same for two reasons:
- the jumplist includes cursor positions even within the same file (sometimes you need to press CTRL-O multiple times to actually jump back to your previous note);
- the jumplist is universal (it's shared across all buffers), while in Obsidian each note has a different "jumplist".

**Comment by kappq** [comment]
I would be willing to build this myself, however I'm not very familiar with Lua nor with how Neovim plugins are built so it will probably take a while.

**Comment by armaninux** [comment]
I second this feature request!

**Comment by sotte** [comment]
I assume `:ObsidianForward` and `:ObsidianBackward` should just cycle through the already opened notes?!

If so, you can just use `:bnext` and `:bprevious` to jump to the next/prev buffer. These commands are [often](https://github.com/tpope/vim-unimpaired) mapped to `]b` and `[b` or similar. 

Alternatively, you can use `:Telescope buffers` to jump to the open buffers.

I think this should solve your issue.

**Comment by kappq** [comment]
> I assume `:ObsidianForward` and `:ObsidianBackward` should just cycle through the already opened notes?!
> 
> If so, you can just use `:bnext` and `:bprevious` to jump to the next/prev buffer. These commands are [often](https://github.com/tpope/vim-unimpaired) mapped to `]b` and `[b` or similar.
> 
> Alternatively, you can use `:Telescope buffers` to jump to the open buffers.
> 
> I think this should solve your issue.

It shouldn't cycle through the already open notes, it should follow (either backwards or forward) the chain of notes that the user created by following various links, which is different from notes opened in different buffers. In fact, when you click on a link in Obsidian, it doesn't actually open it in a new tab, but in place of the current note. This also means that you can get to the same note in various ways.

**Comment by armaninux** [comment]
it's a slightly different thing; bnext and bprev are fixed and depend on the order of access; we're looking instead to navigating the buffers by access history, i.e.
if I open 10 buffers:
buf1 buf2 buf3 buf4 ... buf9 buf10
then access buf4 via a link in buf10, bprev will open buf3, but I want to go back to buf10

**Comment by sotte** [comment]
Ah, I see. Slightly different indeed. Thanks for clarifying.

**Comment by sotte** [comment]
I think this plug-in does what u want: https://github.com/volskaya/windovigation.nvim

**Comment by mscott99** [comment]
I had the same issue and started to use ctrl-O and ctrl-I more. Not quite the same, but usually does the job. And no need for a plugin.

**Comment by 777uliahshafar** [comment]
This is need more development, I am sure many people would love it.

**Comment by hwamil** [comment]
Yeah I'm missing this feature since I've used `vimwiki`'s back feature (mapped to backspace by default).
Was quite handy in quick navigation.

**Comment by sotte** [comment]
Please try [windovigation.nvim](https://github.com/volskaya/windovigation.nvim).

**Comment by 777uliahshafar** [comment]
vim-bufsurf is similar to windovigation, but it is more suitable to my expectation.
